src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding JavaScript runtime based applications to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class JavaScriptAppHostingExtension
+{   
+    /// <summary>
+    /// Adds a JavaScript runtime to the application model. The runtime and dependencies should available on the PATH.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="scriptPath">The path to the script that the runtime will execute.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<JavaScriptAppResource> AddJavaScriptApp(this IDistributedApplicationBuilder builder, string name, string? scriptPath = null, string? workingDirectory = null, string[]? args = null)
+    {   
+        scriptPath ??= "node";
```

Comment:
So if the value of `scriptPath` is not null ... lets say it is something like `myscript.js` - what is expected to happen here. Is the expectation that the JavaScript file itself will have a `#!/bin/...` prefix on it? How would that work on Windows?